xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-macro/xwiki-platform-notifications-macro-ui/src/main/resources/XWiki/Notifications/Code/Macro/NotificationsMacro.xml

```diff
@@ -599,8 +602,11 @@ require(['jquery', 'XWikiAsyncNotificationsMacro', 'xwiki-events-bridge'], funct
           entry.find('.notification-content').prepend(readButton);
           // Style the read button
           readButton.addClass('notification-event-read-button').addClass('btn btn-xs');
-          // Insert the cross icon
-          readButton.html('$services.icon.renderHTML("check")');
+          // Insert the cross icon and an accessible label
+          var readButtonIcon = $('$services.icon.renderHTML("check")');
+          var readButtonLabel= $('&lt;span class="sr-only"/&gt;');
+          readButtonLabel.text('$services.localization.render("notifications.macro.markEventAsRead")');
+          readButton.append(readButtonIcon).append(readButtonLabel);
```

Comment:
Looks like code duplication with above, no? Isn't there a way to prevent the dup?

Reply:
It's a duplication indeed because the page defines several JavaScript extensions, in particular `XWikiNotificationsMacro` and `XWikiAsyncNotificationsMacro` with code duplication. Preventing the duplication is probably doable but not trivial (for me) and might be beyond the scope of this issue (since there is more duplicated code than just for this button generation), what do you think?

Reply:
yes I agree :)